tests/infra/member.py

```diff
@@ -23,22 +23,35 @@ class MemberStatus(Enum):
 
 
 class Member:
-    def __init__(self, member_id, curve, key_generator, common_dir):
+    def __init__(self, member_id, curve, common_dir, key_generator=None):
         self.key_generator = key_generator
         self.common_dir = common_dir
         self.member_id = member_id
         self.status = MemberStatus.ACCEPTED
 
-        # For now, all members are given an encryption key (for recovery)
-        member = f"member{member_id}"
-        infra.proc.ccall(
-            self.key_generator,
-            f"--name={member}",
-            f"--curve={curve.name}",
-            "--gen-enc-key",
-            path=self.common_dir,
-            log_output=False,
-        ).check_returncode()
+        if key_generator is not None:
+            # For now, all members are given an encryption key (for recovery)
+            member = f"member{member_id}"
+            infra.proc.ccall(
+                self.key_generator,
+                f"--name={member}",
+                f"--curve={curve.name}",
+                "--gen-enc-key",
+                path=self.common_dir,
+                log_output=False,
+            ).check_returncode()
+        else:
+            # If no key generator is passed in, the identity of the member
+            # should have been created in advance (e.g. by a previous network)
+            assert os.path.isfile(
```

Comment:
fwiw, pathlib is pretty nice, but this is fine too of course.

Reply:
I've stuck to `os.path` as it is what we've been using throughout the Python infra so far but we can change that in a future PR.